backend/typescript/middlewares/validators/campValidators.ts

```diff
@@ -379,6 +386,56 @@ export const deleteCampSessionsDtoValidator = async (
   return next();
 };
 
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable-next-line import/prefer-default-export */
+export const updateCampSessionsDtoValidator = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  const campSessions = req.body.updatedCampSessions;
+
+  if (!campSessions) {
+    return res.status(400).send("updatedCampSessions is a required field");
+  }
+
+  for (let index = 0; index < campSessions.length; index += 1) {
+    const campSession = campSessions[index];
+    if (!campSession.id) {
+      return res
+        .status(400)
+        .send("id is a required field for all camp sessions");
+    }
+    if (!validatePrimitive(campSession.id, "string")) {
+      return res.status(400).send(getApiValidationError("id", "string"));
+    }
+    if (campSession.dates && !validateArray(campSession.dates, "string")) {
+      return res
+        .status(400)
+        .send(getApiValidationError("dates", "string", true));
+    }
+    if (campSession.dates && !campSession.dates.every(validateDate)) {
+      return res
+        .status(400)
+        .send(getApiValidationError("dates", "Date string"));
+    }
+    if (!validatePrimitive(campSession.capacity, "integer")) {
+      return res.status(400).send(getApiValidationError("capacity", "integer"));
+    }
+    if (campSession.campers) {
```

Comment:
Can we not make this an `UpdateCampSession` type which the body has to conform to? That way, we can never get campers and waitlist in the body (refer to the typing comment above). Might have missed something here though

Reply:
oh, i just saw that you do have that type. can we not just enforce the body to conform to that type above? Maybe that's the point of the validator but otherwise, we would have to check for arbitrary fields right?

Reply:
Yeahh it's not great, I'm wondering if I'm missing some sort of `try { <cast> } catch { return res.status(400) }` pattern? I only check for these because they're fields on the object that may lead to common misuse